src/docs/sdk/performance/ui-event-transactions.mdx

```diff
@@ -0,0 +1,173 @@
+and send the transaction. If not, the SDK should discard the empty transaction. A combination of
+two concepts is needed to implement the wait-and-see logic: __idle transactions__ and
+__wait-for-children__. Check out the specification below to see how those two concepts work
+together with covering multiple edge cases. The following description is drastically simplified:
+
+* __Idle-transactions__: The SDK starts an idle timeout for the transaction when starting it.
+When the transaction or any of its spans starts a new span, it resets the timeout. The SDK finishes
+the transaction when the timeout succeeds.
+* __Wait-for-children__: A transaction waits for all its child spans to finish before finishing itself.
+
+Before diving into the specification with all the edge cases, let's look at two simple examples:
+
+1. The user clicks a button that triggers some requests to the backend and stores data in the
+local database. The SDK can create a meaningful transaction in that case.
+2. The user clicks a button that solely validates some form data and triggers nothing that the
+SDK can automatically instrument. The SDK could only create a transaction without any spans, which would be valueless.
+
+# Specification
+
+Users can change the `idleTimeout` via the SDK config options. The default is 3.0 seconds. 
+
+The specification is written in the [Gherkin syntax](https://cucumber.io/docs/gherkin/reference/).
+
+**1. Starting UI Event transactions** 
+```
+Given an instrumentable UI event
+Then the SDK starts a UI event transaction 
+And starts the idle timeout with idleTimeout of the options
+```
+
+**5. Wait for children when starting span**
+```
+Given a UI event transaction
+When the SDK starts a child span
+Then the SDK cancels the idle timeout
+And waits for the child to finish
+```
+
+**4. Start timeout when the last span finishes**
+```
+Given a UI event transaction
+And the transaction has one or multiple running child spans
+And the transaction is waiting for its children to finish
+When the SDK finishes the last child span
+Then the SDK starts the idle timeout
+```
+
+**3. Don't reset timeout when the second last span finishes**
+```
+Given a UI event transaction
+And the transaction has two running child spans
+When the SDK finishes the first child span
+Then the SDK doesn't reset the idle timeout
+And waits for the children to finish
+```
+
+**2. Discard UI event transactions without child spans**
+```
+Given a UI event transaction
+And the transaction has no child spans
+When the idleTimeout times out
+Then the SDK discards the transaction 
+```
+
+**3. Set time to last finished child span**
+```
+Given a UI event transaction
+And the transaction has one finished child span
+And the transaction has one running child span
+When the running child span finishes
+Then the SDK finishes the transaction
+And trims the end time of the transaction to the one of the last finished child span
+```
```

Comment:
Android and Flutter have an option for that called `trimEnd` and it is set by default for idle transactions.

Reply:
Thanks for the info, but I guess it's OK to keep this? If you have `trimEnd` you are safe.

Reply:
Maybe mention that some SDKs have the `trimEnd`, but you can also skip it and use the timestamp of the last finished child span as well.

Reply:
`trimEnd` what happens on Android if you disable it for that scenario, @romtsn?

Reply:
It won't set the transaction end time to the latest child end time, so it will be set to `latest child end timestamp + idleTimeout`

Reply:
but worth noting, it's not configurable from within SentryOptions, so it's internal on Android and is more of a convenience flag for us if we decide not to trim

Reply:
Yeah, tbh `trimEnd` might be just overengineering, I guess we can skip this.